en/engine/event/event-builtin.md

```diff
@@ -0,0 +1,167 @@
+
+## Touch event propagation
+
+### Touch event bubbles
+
+Touch events support the event bubbles on the node tree, take the pictures below as an example:
+
+![propagation](propagation.png)
+
+In the scene shown in the picture, suppose node A has a child node B which has a child node C. The developer sets the touch event listeners for all these three nodes (each node has a touch event listener in examples below by default).
+
+When the mouse or finger was applied in the node C region, the event will be triggered at node C first and the node C listener will receive the event. Then the node C will pass this event to its parent node, so the node B listener will receive this event. Similarly the node B will also pass the event to its parent node A. This is a basic event bubbling phase. It needs to be emphasized that there is no hit test in parent nodes in the bubbling phase, which means that the node A and B can receive touch events even though the touch location is out of their node region.
+
+The bubbling phase of touch events is no different than the general events. Calling `event.propagationStopped = true;` can force to stop the bubbling phase.
+
+### Ownership of touch points among brother nodes
+
+Suppose the node B and C in the picture above are brother nodes, while C partly covers over B. Now if C receives a touch event, it is announced that the touch point belongs to C, which means that the brother node B won't receive the touch event any more, even though the touch location is also inside its node region. The touch point belongs to the top one among brother nodes.
+
+At the same time, if C has a parent node, it will also pass the touch event to its parent node through the event bubble mechanism.
+
+### Point of Contact Attribution for Different Canvas
+
+Contact interception between different Canvas is determined by priority. In the scene in the figure below, Canvas 1-5 in the node tree corresponds to priority 1-5 of the image display. It can be seen that even though Canvas nodes 3, 4, and 5 are arranged in scrambled order, the order of response of the contacts is still **Canvas5 -> Canvas4 -> Canvas3 -> Canvas2 -> Canvas1**, according to the priority relationship on the Canvas. The sorting between Canvas is done in the order of the node tree only if the priority is the same.
+
+![multi-canvas](multi-canvas.png)
+
+### Register touch or mouse events in the capturing phase
+
+Sometimes it is needed to dispatch the touch or mouse events to parent node event listeners before dispatching to any child nodes beneath it in hierarchy, like the design of __cc.ScrollView__ component.
+
+Now the event bubbling can't meet our demand, so that we need to register the parent node event listeners in the capturing phase. To achieve this goal, passing in `true` as the `useCapture` parameter (a fourth parameter) when registering touch or mouse event on the node. For example:
```

Comment:
Now the event bubbling can't meet our demand, so that we need to register the parent node event listeners in the capturing phase.
-->
Event bubbling cannot meet all demands. When this happens, register the parent node event listeners in the capturing phase.

Reply:
capturing phase

What does this mean?

Reply:
捕获事件所处的阶段（The stage at which the event is captured）